src/state/swap/hooks.ts

```diff
@@ -90,76 +93,121 @@ export function useDerivedSwapInfo(): {
   const tokenIn = useTokenByAddressAndAutomaticallyAdd(tokenInAddress)
   const tokenOut = useTokenByAddressAndAutomaticallyAdd(tokenOutAddress)
 
-  const relevantTokenBalances = useTokenBalancesTreatWETHAsETH(account ?? undefined, [tokenIn, tokenOut])
+  const relevantTokenBalances = useTokenBalancesTreatWETHAsETH(account ?? undefined, [
+    tokenIn ?? undefined,
+    tokenOut ?? undefined
+  ])
 
   const isExactIn: boolean = independentField === Field.INPUT
-  const amount = tryParseAmount(typedValue, isExactIn ? tokenIn : tokenOut)
+  const parsedAmount = tryParseAmount(typedValue, (isExactIn ? tokenIn : tokenOut) ?? undefined)
 
-  const bestTradeExactIn = useTradeExactIn(isExactIn ? amount : undefined, tokenOut)
-  const bestTradeExactOut = useTradeExactOut(tokenIn, !isExactIn ? amount : undefined)
+  const bestTradeExactIn = useTradeExactIn(isExactIn ? parsedAmount : undefined, tokenOut ?? undefined)
+  const bestTradeExactOut = useTradeExactOut(tokenIn ?? undefined, !isExactIn ? parsedAmount : undefined)
 
   const bestTrade = isExactIn ? bestTradeExactIn : bestTradeExactOut
 
-  const parsedAmounts = {
-    [Field.INPUT]: isExactIn ? amount : bestTrade?.inputAmount,
-    [Field.OUTPUT]: isExactIn ? bestTrade?.outputAmount : amount
-  }
-
   const tokenBalances = {
     [Field.INPUT]: relevantTokenBalances?.[tokenIn?.address ?? ''],
     [Field.OUTPUT]: relevantTokenBalances?.[tokenOut?.address ?? '']
   }
 
   const tokens: { [field in Field]?: Token } = {
-    [Field.INPUT]: tokenIn,
-    [Field.OUTPUT]: tokenOut
+    [Field.INPUT]: tokenIn ?? undefined,
+    [Field.OUTPUT]: tokenOut ?? undefined
   }
 
   // get link to trade on v1, if a better rate exists
-  const v1TradeLinkIfBetter = useV1TradeLinkIfBetter(
-    isExactIn,
-    tokens[Field.INPUT],
-    tokens[Field.OUTPUT],
-    isExactIn ? parsedAmounts[Field.INPUT] : parsedAmounts[Field.OUTPUT],
-    bestTrade ?? undefined,
-    V1_TRADE_LINK_THRESHOLD
-  )
+  const v1Trade = useV1Trade(isExactIn, tokens[Field.INPUT], tokens[Field.OUTPUT], parsedAmount)
 
   let error: string | undefined
   if (!account) {
     error = 'Connect Wallet'
   }
 
-  if (!parsedAmounts[Field.INPUT]) {
+  if (!parsedAmount) {
     error = error ?? 'Enter an amount'
   }
 
-  if (!parsedAmounts[Field.OUTPUT]) {
-    error = error ?? 'Enter an amount'
+  if (!tokens[Field.INPUT] || !tokens[Field.OUTPUT]) {
+    error = error ?? 'Select a token'
   }
 
-  const [balanceIn, amountIn] = [tokenBalances[Field.INPUT], parsedAmounts[Field.INPUT]]
-  if (balanceIn && amountIn && balanceIn.lessThan(amountIn)) {
-    error = 'Insufficient ' + amountIn.token.symbol + ' balance'
-  }
+  // this check is incorrect, it should check against the maximum amount in
```

Comment:
this check was basically deleted, users can now approve/swap with insufficient balance, which i think is an ok degradation (not important to solve since this check was wrong).